internal/infrastructure/kubernetes/ratelimit/resource.go

```diff
@@ -185,20 +185,23 @@ func expectedRateLimitContainerEnv(rateLimit *egcfgv1a1.RateLimit, rateLimitDepl
 	}
 
 	if rateLimit.Backend.Redis.TLS != nil {
-		env = append(env, []corev1.EnvVar{
-			{
-				Name:  RedisTLS,
-				Value: "true",
-			},
-			{
-				Name:  RedisTLSClientCertEnvVar,
-				Value: RedisTLSClientCertFilename,
-			},
-			{
-				Name:  RedisTLSClientKeyEnvVar,
-				Value: RedisTLSClientKeyFilename,
-			},
-		}...)
+		env = append(env, corev1.EnvVar{
+			Name:  RedisTLS,
+			Value: "true",
+		})
+
+		if rateLimit.Backend.Redis.TLS.CertificateRef != nil {
+			env = append(env, []corev1.EnvVar{
+				{
+					Name:  RedisTLSClientCertEnvVar,
+					Value: RedisTLSClientCertFilename,
+				},
+				{
+					Name:  RedisTLSClientKeyEnvVar,
+					Value: RedisTLSClientKeyFilename,
```

Comment:
needs ca.crt ? 



https://github.com/envoyproxy/gateway/blob/a7746d648bde075a7b841bc576103ec16a8ee08e/internal/infrastructure/kubernetes/ratelimit/resource.go#L195-L218

Reply:
raised a separate issue to track Ca cert
https://github.com/envoyproxy/gateway/issues/1419

Reply:
oh nvm, we should delete the CACertEnvVar for now

Reply:
this env just exists in past version and removed ref your comment